c/prima.c

```diff
@@ -21,50 +56,127 @@ int lincoa_c(prima_obj calfun, const void *data, const int n, double x[], double
              const double xl[], const double xu[],
              int *nf, const double rhobeg, const double rhoend, const double ftarget, const int maxfun, const int npt, const int iprint, int *info);
 
+int prima_check_options(prima_options * opt, int n)
+{
+  if (!opt)
+    return PRIMA_NULL_OPTIONS;
+  if (!opt->xl)
+  {
+    opt->xl = (double*)malloc(n * sizeof(double));
+    opt->_allocated_xl = 1;
+    for (int i = 0; i< n; ++ i)
+      opt->xl[i] = -HUGE_VAL;
+  }
+  if (!opt->xu)
+  {
+    opt->xu = (double*)malloc(n * sizeof(double));
+    opt->_allocated_xu = 1;
+    for (int i = 0; i< n; ++ i)
+      opt->xu[i] = HUGE_VAL;
+  }
+  if (opt->npt < 0)
+    opt->npt = 2*n+1;
+  return 0;
+}
+
+void prima_free_options(prima_options * opt)
+{
+  if (opt)
+  {
+    if (opt->_allocated_xl)
+    {
+      free(opt->xl);
+      opt->xl = NULL;
```

Comment:
I am not sure, but I think that `opt->xl` should always be set to `NULL` only `free` is called if `opt->_allocated_xl` is true.

Reply:
not necessarily: it can be allocated by the user